src/astro/databases/snowflake.py

```diff
@@ -506,13 +506,23 @@ def schema_exists(self, schema: str) -> bool:
 
         :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
         """
-        created_schemas = [
-            x["SCHEMA_NAME"]
-            for x in self.hook.run(
+
+        # Below code is added due to breaking change in apache-airflow-providers-snowflake==3.2.0,
+        # we need to pass handler param to get the rows. But in version apache-airflow-providers-snowflake==3.1.0
+        # if we pass the handler provider raises an exception AttributeError 'sfid'.
+        try:
+            schemas = self.hook.run(
                 "SELECT SCHEMA_NAME from information_schema.schemata WHERE LOWER(SCHEMA_NAME) = %(schema_name)s;",
                 parameters={"schema_name": schema.lower()},
+                handler=lambda cur: cur.fetchall(),
             )
-        ]
+        except AttributeError:
+            schemas = self.hook.run(
+                "SELECT SCHEMA_NAME from information_schema.schemata WHERE LOWER(SCHEMA_NAME) = %(schema_name)s;",
+                parameters={"schema_name": schema.lower()},
+            )
```

Comment:
Why do we still need this if we upgrade to `3.2.0` ?

Reply:
@feluelle I'm not sure about this part, but I was trying to avoid placing a lower bound on snowflake dependency. To be more backward compatible.

Reply:
@kaxil Thoughts?

Reply:
If we don't wanna support this becomes a lot easier as well :)

Reply:
I would say we should not support prior version (3.1.0).

However if you think or kaxil that it makes sense use `>=3.1.0,<=3.20` or similar to allow installing 3.1.0 if necessary.

Reply:
Yeah, I would prefer to have backwards compatibility as users wouldn't update to 3.2.0 as it was just released a day or two ago.

Reply:
@feluelle 3.2.0 is released recently and not many people will be having the newer version. So I think we should be supporting 3.1.0 as well.